### PR TITLE
New GUI base example for gui team

### LINKED ISSUE #88: Make widget example and starting branch for GUI team

### PR DESCRIPTION
**Pull request by Pmichaud29**
It's the start implementation of the FileTransferWidget.

Generation of UI file is done automatically at compile time and can be done manually by executing 
```bash
rover_gui/ui/generate_ui.sh
```

I believe almost everything has been setup for a ish smooth start for new members

### CODE REVIEW

File: rover_gui/src/QSshFileExplorer/QFileTransferWidget.hpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#include "QSshFileExplorerWidget.hpp"
+
+class QFileTransferWidget : public QWidget
+{
+  public:
+    QFileTransferWidget(QWidget* parent_);
+    ~QFileTransferWidget();
+
+  private:
+    QGridLayout _mainLayout;
```
Comment (Ndumoulin) [inline rover_gui/src/QSshFileExplorer/QFileTransferWidget.hpp:13]: Ca sert à rien mais un splitter pour resizer à la place d'un grid layout serait peut-être approprié ici : 
Screenshot from 2024-10-23 20-35-05

**Review by Ndumoulin — DISMISSED** [review]
Vrm sick ! Pas grand chose à dire, je comprend mieux comment ca va fonctionner maintenant :)

File: rover_gui/src/QSshFileExplorer/QFileTransferWidget.hpp (hunk 1 above)
Comment (Pmichaud29) [reply rover_gui/src/QSshFileExplorer/QFileTransferWidget.hpp:13]: Done

**Review by Ndumoulin — APPROVED** [review]
Excellent ca fonctionne !

**Pull request merged**